spmd/compiler/fusion.py

```diff
@@ -162,22 +214,448 @@ def _scan_graph_for_fusion_elements(
     return element_list
 
 
-def run_comm_fusion(gm: fx.GraphModule) -> bool:
+def _copy_fe_to_buffer(
+    gi: GraphInfo, gm: fx.GraphModule, in_fe_list: list[FusionElement]
+) -> None:
+    """first half of fusion - move desired items to buffer and create graph"""
+    buffer_node = gi.global_buffer_node
+    buffer_size = gi.global_buffer_size
+
+    copy_list = in_fe_list
+
+    num_fusion_elements = len(copy_list)
+
+    def copy_to_buffer(
+        buffer: torch.Tensor, tensor_list: List[torch.Tensor]
+    ) -> torch.Tensor:
+        offset = 0
+        for t in tensor_list:
+            size = t.numel()
+            buffer[offset : offset + size] = t.view(-1)
+            offset += size
+        return buffer
+
+    # setup dummy vars
+    buffer = None
+    if gi.tracing_buffer is None:
+        buffer = torch.empty(buffer_size)
+        gi.tracing_buffer = buffer
+    elif gi.tracing_buffer:
+        buffer = buffer = torch.empty(buffer_size)
+
+    tlist = []
+    for item in copy_list:
+        a = torch.zeros((item.shape[0], item.shape[1]))  # type: ignore
+        tlist.append(a)
+
+    load_gm = make_fx(copy_to_buffer)(buffer, tlist)
+
+    subnodemap = create_graph_node_map(load_gm)
```

Comment:
can we rename this to reflect that this is a mapping from node.name to node?

Reply:
yes - changed name to create_name_to_node_map to make it clearer.